### PR TITLE
Add Patch users

### PR DESCRIPTION
**Pull request by nmacianx**
## Related Issues

Related to https://github.com/descope/etc/issues/6646

## Description

Adds Patch User

## Must

- [X] Tests
- [ ] Documentation (if applicable)

### CODE REVIEW

File: lib/management/user.ts [hunk 1]
```diff
@@ -351,6 +351,63 @@ const withUser = (sdk: CoreSdk, managementKey?: string) => {
...
+      body.verifiedEmail = options.verifiedEmail;
+    }
+    if (options.verifiedPhone !== undefined) {
+      body.verifiedPhone = options.verifiedPhone;
+    }
+    if (options.ssoAppIds !== undefined) {
+      body.ssoAppIds = options.ssoAppIds;
+    }
+
+    // TODO patch instead of post
```
Comment (nmacianx) [inline lib/management/user.ts:404]: TODO

**Comment by nmacianx** [comment]
@dorsha this can't be merged since updating the Core JS SDK breaks Node SDK

File: lib/fetch-polyfill.ts [hunk 2]
```diff
@@ -9,8 +9,11 @@ const patchedFetch = (...args: Parameters<typeof crossFetch>) => {
   // we can get Request on the first arg, or RequestInfo on the second arg
   // we want to make sure we are setting the "highWaterMark" so we are doing it on both args
   args.forEach((arg) => {
-    // eslint-disable-next-line no-param-reassign, @typescript-eslint/no-unused-expressions
-    arg && ((arg as any).highWaterMark ??= highWaterMarkMb);
+    // Updated to only apply highWaterMark to objects, as it can't be applied to strings (it breaks it)
+    if (arg && typeof arg === 'object') {
+      // eslint-disable-next-line no-param-reassign, @typescript-eslint/no-unused-expressions
+      (arg as any).highWaterMark ??= highWaterMarkMb;
+    }
```
Comment (nmacianx) [inline lib/fetch-polyfill.ts:16]: FYI this is the fix for the broken SDK
@asafshen @itaihanski @nirgur

Comment (nmacianx) [reply lib/fetch-polyfill.ts:16]: In a PR about 2 months ago, a change was introduced to the core-sdk where one of the params used to call `fetch` was changed from URL to a string, and it breaks here

**Review by dorsha — APPROVED** [review]

**Review by dorsha — APPROVED** [review]

**Pull request merged**